src/ViewLayers/ViewLayerPath.vala

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2019-2021 Alecaddd (https://alecaddd.com)
+ *
+ * This file is part of Akira.
+ *
+ * Akira is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+
+ * Akira is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+
+ * You should have received a copy of the GNU General Public License
+ * along with Akira. If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Authored by: Ashish Shevale <shevaleashish@gmail.com>
+ */
+
+public class Akira.ViewLayers.ViewLayerPath : ViewLayer {
+    public const double UI_NOB_SIZE = 4;
+    public const double UI_LINE_WIDTH = 1.01;
+
+    private Gee.ArrayList<Geometry.Point?>? points = null;
+    private Geometry.Rectangle extents;
+    private Geometry.Point first_point;
+
+    public void set_reference_point (Geometry.Point reference_point) {
+        first_point = reference_point;
+        canvas.request_redraw (extents);
```

Comment:
it's better to create an "update" override that redraws the extents before and after an update. We can keep it to only update the new extent, but still creating that update method would be better.